src/java/com/sarality/app/view/dialog/CustomViewDialogFragment.java

```diff
@@ -33,10 +33,14 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
   public void onActivityCreated(Bundle savedInstanceState) {
     super.onActivityCreated(savedInstanceState);
     dataSource = createDataSource();
-    if (dataSource == null) {
-      render(null);
-    } else {
+    init();
```

Comment:
I don't think you need to do all this.

Just add a method

public void init(T data) {
   render(data);
}

Are there places where you need to call this Dialog where you don't have this data available i.e. you need to call init without any parameters and it needs to reuse the data that it was initialized with? If not, then the init(T data) approach should work just fine.